tests/ui/test_ui_tools.py

```diff
@@ -2780,28 +2780,177 @@ def test_transform_content(self, mocker, raw_html, expected_content):
+                            "email": "AARON@zulip.com",
+                            "full_name": "aaron",
+                            "id": 1,
+                        },
+                        "reaction_type": "zulip_extra_emoji",
+                    },
+                    {
+                        "emoji_name": "zulip",
+                        "emoji_code": "zulip",
+                        "user": {
+                            "email": "shivam@zulip.com",
+                            "full_name": "Shivam",
+                            "id": 6,
+                        },
+                        "reaction_type": "unicode_emoji",
+                    },
+                ],
+                "expected_result": " :heart: Iago   :zulip: Shivam, You  ",
+                "reaction_attributes": [
+                    ("reaction", 14),
+                    (None, 1),
+                    ("reaction_mine", 21),
+                ],
+            },
+            {
+                "reactions": [
+                    {
+                        "emoji_name": "zulip",
+                        "emoji_code": "zulip",
+                        "user": {
+                            "email": "AARON@zulip.com",
+                            "full_name": "aaron",
+                            "id": 1,
+                        },
+                        "reaction_type": "zulip_extra_emoji",
+                    },
+                    {
+                        "emoji_name": "zulip",
+                        "emoji_code": "zulip",
+                        "user": {
+                            "email": "shivam@zulip.com",
+                            "full_name": "Shivam",
+                            "id": 6,
+                        },
+                        "reaction_type": "unicode_emoji",
+                    },
+                ],
+                "expected_result": " :zulip: Shivam, You  ",
+                "reaction_attributes": [
+                    ("reaction_mine", 21),
+                ],
+            },
         ],
     )
     def test_reactions_view(self, message_fixture, to_vary_in_each_message):
         self.model.user_id = 1
         varied_message = dict(message_fixture, **to_vary_in_each_message)
         msg_box = MessageBox(varied_message, self.model, None)
         reactions = to_vary_in_each_message["reactions"]
-
+        expected_result = to_vary_in_each_message["expected_result"]
+        reaction_attributes = to_vary_in_each_message["reaction_attributes"]
         reactions_view = msg_box.reactions_view(reactions)
 
-        assert reactions_view.original_widget.text == (
-            " :heart: 1   :thumbs_up: 1   :zulip: 2  "
-        )
-        assert reactions_view.original_widget.attrib == [
-            ("reaction", 11),
-            (None, 1),
-            ("reaction", 15),
-            (None, 1),
-            ("reaction_mine", 11),
-        ]
+        assert reactions_view.original_widget.text == (expected_result)
```

Comment:
Is this a left-over parenthesis? Black reformats code, but please check the code for readability afterwards :)